server/py/services/api/runtime_handlers/mpijob/abstract.py

```diff
@@ -50,37 +52,28 @@ def run(
 
         job = self._generate_mpi_job(runtime, run, execution, meta)
 
-        resp = self._submit_mpijob(job, meta.namespace)
-
-        state = None
-        timeout = int(config.submit_timeout) or 120
-        for _ in range(timeout):
-            resp = self.get_job(meta.name, meta.namespace)
-            state = self._get_job_launcher_status(resp)
-            if resp and state:
-                break
-            time.sleep(1)
-
-        if resp:
-            logger.info(f"MpiJob {meta.name} state={state or 'unknown'}")
-            if state:
-                state = self._crd_state_to_run_state(state)
-                launcher, _ = self._get_launcher(meta.name, meta.namespace)
-                execution.set_hostname(launcher)
-                execution.set_state(state)
-                txt = f"MpiJob {meta.name} launcher pod {launcher} state {state}"
-                logger.info(txt)
-                run.status.status_text = txt
-
-            else:
-                pods_phases = self.get_pods(meta.name, meta.namespace)
-                txt = f"MpiJob status unknown or failed, check pods: {pods_phases}"
-                logger.warning(
-                    "MpiJob status unknown or failed",
-                    pods_phases=pods_phases,
-                    resp_status=mlrun.utils.get_in(resp, "status"),
-                )
-                run.status.status_text = txt
+        self._submit_mpijob(job, meta.namespace)
+
+        # fetch the launcher pod status
+        resp = self.get_job(meta.name, meta.namespace)
+        status = self._get_job_launcher_status(resp)
+
+        if status:
+            # map the CRD state to run state and set hostname if launcher started
+            state = self._crd_state_to_run_state(status)
+            launcher, _ = self._get_launcher(meta.name, meta.namespace)
+            execution.set_hostname(launcher)
+            txt = f"MpiJob {meta.name} launcher pod {launcher} is in state {state}"
+            logger.info(txt)
+        else:
+            # no state yet, assume pending
+            state = mlrun.run.RunStatuses.pending
+            txt = f"MpiJob {meta.name} pending - awaiting launcher pod startup"
+            logger.info(txt)
```

Comment:
```suggestion
            logger.info("Waiting for MpiJob launcher pod to start", name=meta.name, state=state)
```
structured log is preferred, giving more organized and context-based log lines, same goes to line above